### PR TITLE
Issue 41

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: chain/output.go [hunk 1]
```diff
@@ -8,9 +8,10 @@ import (
 	"math/big"
 )
 
+// JSON tags needed for test fixtures
 type Output struct {
-	NewOwner common.Address
-	Amount   *big.Int
+	NewOwner common.Address `json:"NewOwner"`
+	Amount   *big.Int       `json:"NewOwner"`
```
Comment (mslipper) [inline chain/output.go]: Duplicate?

File: chain/transaction.go [hunk 2]
```diff
@@ -185,56 +171,56 @@ func (tx *Transaction) RLPHash() util.Hash {
...
+	return rlp.Encode(w, &itf)
 }
+
+func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
+	var itf rlpHelper
+	err := s.Decode(&itf)
+	if err != nil {
+		return err
+	}
+	tx.Input0  = ZeroInput()
```
Comment (mslipper) [inline chain/transaction.go]: Could leverage constructor pattern for these:

`tx.input0 = NewInput(blkNum0, txIdx0)`

File: eth/plasma.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ type PlasmaClient struct {
 	plasma      *contracts.Plasma
 	privateKey  *ecdsa.PrivateKey
 	userAddress string
-	ethClient   *Client
+	ethClient   Client
```
Comment (mslipper) [inline eth/plasma.go]: Won't this copy the eth client?

File: util/hash.go [hunk 4]
```diff
@@ -14,6 +17,20 @@ type RLPHashable interface {
 	RLPHash() Hash
 }
 
+// Custom JSON unmarshal needed for test fixtures
+func (h *Hash) UnmarshalJSON(data []byte) error {
+	s := string(data[1:len(data)-1])
```
Comment (mslipper) [inline util/hash.go]: I'm assuming that this goes from a `0x` hex string to bytes?

File: eth/plasma.go (hunk 3 above)
Comment (gcristea) [reply eth/plasma.go]: Not ethClient is an interface (from a struct).

File: util/hash.go (hunk 4 above)
Comment (gcristea) [reply util/hash.go]: Not exactly, the json parser sends the entire field (string quotes included.)

File: chain/output.go (hunk 1 above)
Comment (gcristea) [reply chain/output.go]: Fixed.

File: chain/transaction.go (hunk 2 above)
Comment (gcristea) [reply chain/transaction.go]: Fixed.

**Pull request merged**